crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -1121,17 +1106,72 @@ impl actor::HcP2p for HolochainP2pActor {
 
     fn publish(
         &self,
-        _dna_hash: DnaHash,
-        _request_validation_receipt: bool,
-        _countersigning_session: bool,
-        _basis_hash: holo_hash::OpBasis,
+        dna_hash: DnaHash,
+        request_validation_receipt: bool,
+        countersigning_session: bool,
+        basis_hash: holo_hash::OpBasis,
         _source: AgentPubKey,
-        _op_hash_list: Vec<DhtOpHash>,
+        op_hash_list: Vec<DhtOpHash>,
         _timeout_ms: Option<u64>,
-        _reflect_ops: Option<Vec<DhtOp>>,
+        reflect_ops: Option<Vec<DhtOp>>,
     ) -> BoxFut<'_, HolochainP2pResult<()>> {
         Box::pin(async move {
-            tracing::error!("publish is currently a STUB in holochain_p2p--deferring since we'll eventually get stuff on gossip anyways...");
+            use crate::types::event::HcP2pHandler;
+
+            // This single function is/was a weird mix of a bunch of strange
+            // operations.
+            //
+            // - inexplicably `reflect_ops` back to holochain
+            // - actually publish `op_hash_list`
+            // - send a notification to peers that we'd like validation receipts
+            //   - moving this to a separate call
+            // - i have no idea at all what the countersigning bool is for
+
+            // -- handle the bizzarre reflection thing -- //
+
+            if let Some(reflect_ops) = reflect_ops {
+                self.evt_sender
+                    .get()
+                    .ok_or_else(|| HolochainP2pError::other(EVT_REG_ERR))?
+                    .handle_publish(
+                        dna_hash.clone(),
+                        request_validation_receipt,
+                        countersigning_session,
+                        reflect_ops,
+                    )
+                    .await?;
+            }
```

Comment:
Does it break anything if we're not doing this reflecting thing here?

Reply:
No way to know at the moment since so many holochain tests are still broken for other reasons.

Reply:
I'd say leave a TODO and if this isn't being used by any tests then we just remove it. I can't think of a reason outside of testing why we'd send ops back to ourselves. We have too many copies of our data, not too few :)

Reply:
Added todo: https://github.com/holochain/holochain/pull/4788/commits/fd9085e8c209265ba252a69662a82281fce935d1